include/net/azure_iot_hub.h

```diff
@@ -30,6 +30,12 @@ enum azure_iot_hub_evt_type {
 	AZURE_IOT_HUB_EVT_CONNECTING = 0x1,
 	/** Connected to Azure IoT Hub. */
 	AZURE_IOT_HUB_EVT_CONNECTED,
+	/** Connection attempt failed. The reported error code from the
+	 *  IoT Hub is available in the data.err member in the event structure.
+	 *  The error codes correspond to return codes in MQTT CONNACK
```

Comment:
nit: error code in singular form. The event only carries one error code?

Reply:
I thought about that, and found that it made sense this way, as I refer to the set of possible codes, rather than this one in particualr. But could be made for clear if desired